src/core/mesh/qgsmeshlayer.h

```diff
@@ -221,6 +221,20 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
      */
     void setTimeSettings( const QgsMeshTimeSettings &settings );
 
+    /**
+     * Sets time default format settings
+     *
+     * \since QGIS 3.12
+     */
+    void setDefaultTimeSettings();
```

Comment:
this function probably does not fir into qgsmeshlayer class. I would create some helper (ideally not exported out of CORE) function somewhere in mesh layer utils or the class that calls setTimeSettings. so the flow would be

1) resolve refence time
2) assign to QgsMeshTimeSettings
3) call meshlayer->setTimeSettings

Reply:
ok, I simplified that with only a new method in qgsmeshutils. This method return the first valid reference time of the dataset goups.
This function is called from the QgsMeshTimeFormatDialog (only when needed)  but also from the QgsMeshLayer class just after the provider is set. I think this last call is the best place to set the default reference time for those reasons :
- default reference time is defined when the mesh layer is created. Whatever needs the reference time (time widget, active renderer widget, other future stuff), the default reference time is defined from the layer's data.
- When the mesh layer is created when opening project, the read xml method can replace the default time reference with the one saved